src/mlpack/methods/ann/layer/gru_impl.hpp

```diff
@@ -76,6 +76,59 @@ GRU<InputDataType, OutputDataType>::GRU(
   gradIterator = outParameter.end();
 }
 
+// Copy Constructor definition.
+template <typename InputDataType, typename OutputDataType>
+GRU<InputDataType, OutputDataType>::GRU(const GRU& layer) :
+    inSize(layer.inSize),
+    outSize(layer.outSize),
+    rho(layer.rho),
+    weights(layer.weights)
```

Comment:
Could you try including batch_size, forward_step, backward_step, gradient_step and other parameters in the initialisation list ? I believe that should solve the segmentation fault.

Reply:
Thanks @Abilityguy 
I have made the changes that you have suggested
Hopefully, the code does not throw the same error this time